src/Symfony/Component/RemoteEvent/Event/Mailer/AbstractMailerEvent.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\RemoteEvent\Event\Mailer;
+
+use Symfony\Component\RemoteEvent\Event;
+
+/**
+ * @author Fabien Potencier <fabien@symfony.com>
+ *
+ * @experimental in 6.3
+ */
+abstract class AbstractMailerEvent extends Event
+{
+    private \DateTimeImmutable $date;
+    private string $email = '';
+    private array $metadata = [];
+    private array $tags = [];
+
+    public function setDate(\DateTimeImmutable $date): void
```

Comment:
Should not those events be immutable ? 
As they are can be sent by third parties, it does not make sense IMHO to allow to change an incoming event.
However, that class could be used for outgoing events. So modification could be needed even though I think immutability should be a good practice in that case.

Reply:
The date is already immutable

Reply:
The setter makes the event mutable

Reply:
Oh my bad, I thought it was about the `$date` type.